runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java

```diff
@@ -84,9 +86,22 @@
   private MetricsMap<KV<MetricName, HistogramData.BucketType>, HistogramCell> histograms =
       new MetricsMap<>(HistogramCell::new);
 
-  /** Create a new {@link MetricsContainerImpl} associated with the given {@code stepName}. */
+  /**
+   * Create a new {@link MetricsContainerImpl} associated with the given {@code stepName}. If
+   * stepName is null, this MetricsContainer is not bound to a step.
```

Comment:
Is not being bound to a step equivalent to being process wide?

Reply:
No. its not equivalent. We have these in a some other places I believe.


It was used here as well
https://github.com/apache/beam/blob/8463a054c1d7e2b7ee8d11e9569e065cb5e02196/runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerStepMap.java#L52

I believe this was added for some metrics which had the step already in the label information already. (In some code which was tracking metrics for multiple steps). So there was no need to have the container know about the step name.

Likely one of the system metrics. ElementCount, ExecutionTime, etc.

If you feel this needs to be changed, please pick a specific option and I will introduce it. One I can think of is:
adding static factory method (which will set the step to null and isProcessWide to true):
public static MetricsContainerImpl createProcessWideContainer();

Reply:
It feels like that if MetricContainer[Impls] are only sometimes associated with steps, the step association should be stored elsewhere, not as part of this instance. But maybe that's too big a refactoring for now. 

+1 to `MetricsContainerImpl.createProcessWideContainer()`; the nullary constructor doesn't make it obvious the intent is to have this process-wide.

Reply:
Added MetricsContainerImpl.createProcessWideContainer()